### DISCUSSION TITLE
HAA configuration

### DISCUSSION THREAD

**Discussion by mafyata** [RavenSystem/esp-homekit-devices#2010] [MEPLHAA Scripts]
Hi. I updated one device for test new version of HAA, but can’t configurate the script for HAA v12.
Device is Magic Home RGB controller.
This is my script, working on HAA v11:
{"c":{"n":"Hol-Led","b":[{"g":0,"t":0}]},"a":[{"t":30,"s":5,"g":[5,12,13],"fr":5,"fg":5,"fv":5},{"0":{"r":[]},"1":{"r":[],"s":[{"a":1}]},"t":1,"b":[],"s":0}]}
Can you help me to adapt the script for HAA v12?

**Comment by RavenSystem** [comment]
```json
{"c":{"io":[[[0],6],[[5,12,13],7]],"n":"Hol-Led","b":[{"g":0,"t":0}]},"a":[{"t":30,"s":5,"g":[5,12,13]},{"i":0.5,"1":{"s":[{"a":1}]}}]}
```

**Reply by mafyata** [reply]
Thank you very much.

**Comment by mafyata** [comment]
Can you convert these 2
"c":{"l":13,"n":"Lampion","b":[{"g":0,"t":5}]},"a":[{"s":5,"0":{"r":[{"g":12}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0}]},{"0":{"r":[]},"1":{"r":[],"s":[{"a":1}]},"t":1,"b":[],"s":0}]}

{"c":{"l":13,"n":"Fontan","b":[{"g":0,"t":5}]},"a":[{"t": 20,"0":{"r":[{"g":12,"v":0}]},"1":{"r":[{"g":12,"v":1}]},"b":[{"g":0,"t":1}],"s":5,"w":1,"d":0},{"0":{"r":[]},"1":{"r":[],"s":[{"a": 1}]},"t":1,"b":[],"s":0}]}

**Reply by xrust83** [reply]
99% Sonoff Basic. …”io”:[[[13],3],[[12],2],[[0],6]],…

**Reply by mafyata** [reply]
Yes sonoff basic. I can’t compile full script
